test/e2e/nodepool_prev_release_test.go

```diff
@@ -36,6 +37,11 @@ func (npPrevTest *NodePoolPrevReleaseCreateTest) Setup(t *testing.T) {
 }
 
 func (npPrevTest *NodePoolPrevReleaseCreateTest) BuildNodePoolManifest(defaultNodepool hyperv1.NodePool) (*hyperv1.NodePool, error) {
+	isCompatible, err := e2eutil.ValidateNodePoolVersionCompatibility(npPrevTest.release, npPrevTest.hostedCluster.Spec.Release.Image)
```

Comment:
why do we need this check here?

Reply:
I’ve refactored this part of the code and moved it to Setup().

Reply:
This might produce false passes freely since it would transparently skip any test where this is not met?
Why would we want to do this at all? If a test creates a nodepool that breaks the policy, the test should legitimately fail the conditions check and fail